dkc/core/rest/folder.py

```diff
@@ -134,16 +150,17 @@ def get_serializer_context(self):
     # Atomically roll back the tree creation if folder creation fails
     @transaction.atomic
     def perform_create(self, serializer: serializers.ModelSerializer):
-        parent: Folder = serializer.validated_data.get('parent')
+        parent: Folder = serializer.validated_data['parent']
+        user: User = self.request.user
         if parent:
             tree = parent.tree
-            if not tree.has_permission(serializer.context['user'], permission=Permission.write):
```

Comment:
Serializer context seems to be intended to provide additional data *into* the serializer, and not for pulling other data *out* of it (that's the role of `HiddenField`).

In this method, we have more direct access to the authoritative `self.request.user` anyway.